zkvm/src/types.rs

```diff
@@ -329,6 +375,36 @@ impl Data {
             },
         }
     }
+
+    /// Encodes blinded Data values for txprogram bytecode.
+    pub fn encode(&self, program: &mut Vec<u8>) {
+        match self {
+            Data::Opaque(x) => {
+                program.append(&mut x.clone());
+                return;
+            }
+            Data::Witness(w) => w.encode(program),
+        };
+    }
+}
+
+impl DataWitness {
+    fn encode(&self, program: &mut Vec<u8>) {
```

Comment:
I'd rename all `encode` methods from using `program` to `buf` to be a bit more agnostic of where the bytes go.